csharp/src/Ice/Instance.cs

```diff
@@ -753,7 +758,7 @@ private static string typeToClass(string id)
         //
         // Only for use by Ice.CommunicatorI
         //
-        public Instance(Ice.Communicator communicator, Ice.InitializationData initData)
+        internal void initialize(Ice.Communicator communicator, Ice.InitializationData initData)
```

Comment:
because the constructor ended up calling getCommunicator, to get client auth options, which would be null. I think we should merge CommunicatorI and Instance in a follow up PR. And maybe Communicator too?